### PR TITLE
[connectivity][connectvity_platform_interface] migrate to nnbd

### PR DESCRIPTION
**Pull request by cyanglaz**
Since connectivity_web has not migrated, temporarily un-endorse the web plugin.

All the deps are path deps for now, will submit another PR to update them to git dep after landing this.

### CODE REVIEW

File: packages/connectivity/connectivity/lib/connectivity.dart [hunk 1]
```diff
@@ -23,12 +23,12 @@ class Connectivity {
     if (_singleton == null) {
       _singleton = Connectivity._();
     }
-    return _singleton;
+    return _singleton!;
```
Comment (blasten) [inline packages/connectivity/connectivity/lib/connectivity.dart:26]: Interesting. In some cases the analyzer is smart enough to figure out that `_singleton` cannot be `null` at this point, so `!` isn't required.

File: packages/connectivity/connectivity_platform_interface/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 1.1.0-nullsafety
```
Comment (blasten) [inline packages/connectivity/connectivity_platform_interface/CHANGELOG.md:1]: should this be `2.0.0-nullsafety` ?

File: packages/connectivity/connectivity_platform_interface/lib/src/utils.dart [hunk 3]
```diff
@@ -14,7 +14,7 @@ ConnectivityResult parseConnectivityResult(String state) {
 }
 
 /// Convert a String to a LocationAuthorizationStatus value.
-LocationAuthorizationStatus parseLocationAuthorizationStatus(String result) {
+LocationAuthorizationStatus parseLocationAuthorizationStatus(String? result) {
```
Comment (blasten) [inline packages/connectivity/connectivity_platform_interface/lib/src/utils.dart:17]: looks like `result` could be non-null, right?

File: packages/connectivity/connectivity_platform_interface/lib/src/utils.dart [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import 'package:connectivity_platform_interface/connectivity_platform_interface.dart';
 
 /// Convert a String to a ConnectivityResult value.
-ConnectivityResult parseConnectivityResult(String state) {
+ConnectivityResult parseConnectivityResult(String? state) {
```
Comment (blasten) [inline packages/connectivity/connectivity_platform_interface/lib/src/utils.dart:4]: looks like `state` could be non-null, right?

**Review by blasten — APPROVED** [review]
LGTM % minor comments

File: packages/connectivity/connectivity_platform_interface/CHANGELOG.md (hunk 2 above)
Comment (cyanglaz) [reply packages/connectivity/connectivity_platform_interface/CHANGELOG.md:1]: I saw that other platform_interfaces are only bumping minor version. Are we going to break platform_interface as well?

File: packages/connectivity/connectivity_platform_interface/lib/src/utils.dart (hunk 4 above)
Comment (cyanglaz) [reply packages/connectivity/connectivity_platform_interface/lib/src/utils.dart]: done

File: packages/connectivity/connectivity_platform_interface/lib/src/utils.dart (hunk 3 above)
Comment (cyanglaz) [reply packages/connectivity/connectivity_platform_interface/lib/src/utils.dart]: done

File: packages/connectivity/connectivity/lib/connectivity.dart (hunk 1 above)
Comment (cyanglaz) [reply packages/connectivity/connectivity/lib/connectivity.dart:25]: Got an error trying to remove !
```
./lib/connectivity.dart:25:12: Error: A value of type 'Connectivity?' can't be returned from a function with return type 'Connectivity'.
 - 'Connectivity' is from 'package:connectivity/connectivity.dart' ('../lib/connectivity.dart').
    return _singleton;```

**Review by cyanglaz — COMMENTED** [review]
PTAL

**Pull request merged**